src/renderer_server/event_loop_notifier.rs

```diff
@@ -37,7 +39,11 @@ pub struct EventLoopNotifier {
 
 impl EventLoopNotifier {
     pub fn new(event_loop: EventLoopProxy<MainThreadAction>) -> Self {
-        Self {event_loop}
+        Self { event_loop }
+    }
+
+    pub fn exit(&self) -> Result<(), EventLoopClosed> {
```

Comment:
nit: could you move this method down to the bottom so it matches the order in the enum? (makes it a bit easier to see that each variant of the enum has a corresponding method here)